core/src/validator.rs

```diff
@@ -249,6 +250,48 @@ impl TransactionStructure {
     }
 }
 
+#[derive(
+    Clone, Debug, EnumVariantNames, IntoStaticStr, Display, Serialize, Deserialize, PartialEq, Eq,
+)]
+#[strum(serialize_all = "kebab-case")]
+#[serde(rename_all = "kebab-case")]
+pub enum SchedulerPacing {
+    Disabled,
+    FillTimeMillis(NonZeroU64),
+}
+
+impl Default for SchedulerPacing {
+    fn default() -> Self {
+        Self::FillTimeMillis(BankingStage::default_fill_time_millis())
+    }
+}
+
+impl FromStr for SchedulerPacing {
+    type Err = String;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        if s.eq_ignore_ascii_case("disabled") {
+            Ok(SchedulerPacing::Disabled)
+        } else {
+            match s.parse::<u64>() {
+                Ok(v) if v > 0 => Ok(SchedulerPacing::FillTimeMillis(
+                    NonZeroU64::new(v).ok_or_else(|| "value must be non-zero".to_string())?,
+                )),
+                _ => Err("value must be a positive integer or 'disabled'".to_string()),
+            }
+        }
+    }
+}
+
+impl From<&SchedulerPacing> for Option<Duration> {
+    fn from(pacing: &SchedulerPacing) -> Self {
+        match pacing {
+            SchedulerPacing::Disabled => None,
+            SchedulerPacing::FillTimeMillis(millis) => Some(Duration::from_millis(millis.get())),
+        }
+    }
+}
+
```

Comment:
i think something like `fn fill_time(&self) -> Option<Duration>` would be more self-descriptive than a `From` impl here